### ISSUE TITLE
Whiteout outside of rootfs error: /etc/localtime

### ISSUE THREAD

**Issue by gnguy** [apptainer/singularity#1576]
I just ran into this behavior when upgrading to Singularity 2.5 -- I assume it is related to the whiteout extraction that was brought in in 2.5.0, but not sure exactly what might be causing this issue (or if it is intended behavior).

## Version of Singularity:
2.5.1

### Expected behavior
When converting a Docker image that includes modifications to /etc/localtime, the Docker image should be converted to a Singularity image without error.

### Actual behavior
```
/share/local/singularity-2.5.1/bin/singularity --debug build {img_file_name} {registry_image_name}
... {skipping debug output for intermediate layers etc.}
Exploding layer: sha256:f92dd12ef7b10d448dbaf89bcee767aafc0f15dadde6008cf4bc652b86dab715.tar.gz
... {skipping registry output}
DEBUG   [U=70179,P=14755]  main()                                    ROOTFS /tmp/.singularity-build.DAwXW3 canonicalized to /tmp/.singularity-build.DAwXW3
DEBUG   [U=70179,P=14755]  main()                                    Applying whiteouts for tar file /snfs2/HOME/gngu/.singularity/docker/sha256:f92dd12ef7b10d448dbaf89bcee767aafc0f15dadde6008cf4bc652b86dab715.tar.gz
DEBUG   [U=70179,P=14755]  apply_whiteouts()                         Whiteout Marker etc/.wh.localtime
ERROR   [U=70179,P=14755]  apply_whiteout()                          Attempt to whiteout outside of rootfs /usr/share/zoneinfo/Etc/UTC - aborting.
ABORT   [U=70179,P=14755]  apply_whiteout()                          Retval = 255
Cleaning up...
```

### Steps to reproduce behavior

- Build docker image with the following Dockerfile:

```
FROM debian:stretch
RUN rm /etc/localtime
RUN cp /usr/share/zoneinfo/America/Los_Angeles /etc/localtime
```

- Push image to Docker registry

- Run `/share/local/singularity-2.5.1/bin/singularity --debug build {img_file_name} {registry_image_name}`

**Comment by dtrudg** [comment]
Hi @gnguy - thank you for reporting this. I'll need to work up a patch for the docker extraction code to handle the fact that we have a whiteout against an absolute symlink here.

**Comment by dtrudg** [comment]
Closing as fixed via #1605 - if you experience this on latest Singularity please feel free to reopen.
